// marked/lib/marked.cjs
/**
 * marked v4.3.0 - a markdown parser
 * Copyright (c) 2011-2023, Christopher Jeffrey. (MIT Licensed)
 * https://github.com/markedjs/marked
 */

/**
 * DO NOT EDIT THIS FILE
 * The code in this file is generated from files in ./src/
 */

'use strict';

function _defineProperties(target, props) {
  for (var i = 0; i < props.length; i++) {
    var descriptor = props[i];
    descriptor.enumerable = descriptor.enumerable || false;
    descriptor.configurable = true;
    if ("value" in descriptor) descriptor.writable = true;
    Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor);
  }
}
function _createClass(Constructor, protoProps, staticProps) {
  if (protoProps) _defineProperties(Constructor.prototype, protoProps);
  if (staticProps) _defineProperties(Constructor, staticProps);
  Object.defineProperty(Constructor, "prototype", {
    writable: false
  });
  return Constructor;
}
function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}
function _unsupportedIterableToArray(o, minLen) {
  if (!o) return;
  if (typeof o === "string") return _arrayLikeToArray(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor) n = o.constructor.name;
  if (n === "Map" || n === "Set") return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen);
}
function _arrayLikeToArray(arr, len) {
  if (len == null || len > arr.length) len = arr.length;
  for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i];
  return arr2;
}
function _createForOfIteratorHelperLoose(o, allowArrayLike) {
  var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"];
  if (it) return (it = it.call(o)).next.bind(it);
  if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") {
    if (it) o = it;
    var i = 0;
    return function () {
      if (i >= o.length) return {
        done: true
      };
      return {
        done: false,
        value: o[i++]
      };
    };
  }
  throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}
function _toPrimitive(input, hint) {
  if (typeof input !== "object" || input === null) return input;
  var prim = input[Symbol.toPrimitive];
  if (prim !== undefined) {
    var res = prim.call(input, hint || "default");
    if (typeof res !== "object") return res;
    throw new TypeError("@@toPrimitive must return a primitive value.");
  }
  return (hint === "string" ? String : Number)(input);
}
function _toPropertyKey(arg) {
  var key = _toPrimitive(arg, "string");
  return typeof key === "symbol" ? key : String(key);
}

function getDefaults() {
  return {
    async: false,
    baseUrl: null,
    breaks: false,
    extensions: null,
    gfm: true,
    headerIds: true,
    headerPrefix: '',
    highlight: null,
    hooks: null,
    langPrefix: 'language-',
    mangle: true,
    pedantic: false,
    renderer: null,
    sanitize: false,
    sanitizer: null,
    silent: false,
    smartypants: false,
    tokenizer: null,
    walkTokens: null,
    xhtml: false
  };
}
exports.defaults = getDefaults();
function changeDefaults(newDefaults) {
  exports.defaults = newDefaults;
}

/**
 * Helpers
 */
var escapeTest = /[&<>"']/;
var escapeReplace = new RegExp(escapeTest.source, 'g');
var escapeTestNoEncode = /[<>"']|&(?!(#\d{1,7}|#[Xx][a-fA-F0-9]{1,6}|\w+);)/;
var escapeReplaceNoEncode = new RegExp(escapeTestNoEncode.source, 'g');
var escapeReplacements = {
  '&': '&amp;',
  '<': '&lt;',
  '>': '&gt;',
  '"': '&quot;',
  "'": '&#39;'
};
var getEscapeReplacement = function getEscapeReplacement(ch) {
  return escapeReplacements[ch];
};
function escape(html, encode) {
  if (encode) {
    if (escapeTest.test(html)) {
      return html.replace(escapeReplace, getEscapeReplacement);
    }
  } else {
    if (escapeTestNoEncode.test(html)) {
      return html.replace(escapeReplaceNoEncode, getEscapeReplacement);
    }
  }
  return html;
}
var unescapeTest = /&(#(?:\d+)|(?:#x[0-9A-Fa-f]+)|(?:\w+));?/ig;

/**
 * @param {string} html
 */
function unescape(html) {
  // explicitly match decimal, hex, and named HTML entities
  return html.replace(unescapeTest, function (_, n) {
    n = n.toLowerCase();
    if (n === 'colon') return ':';
    if (n.charAt(0) === '#') {
      return n.charAt(1) === 'x' ? String.fromCharCode(parseInt(n.substring(2), 16)) : String.fromCharCode(+n.substring(1));
    }
    return '';
  });
}
var caret = /(^|[^\[])\^/g;

/**
 * @param {string | RegExp} regex
 * @param {string} opt
 */
function edit(regex, opt) {
  regex = typeof regex === 'string' ? regex : regex.source;
  opt = opt || '';
  var obj = {
    replace: function replace(name, val) {
      val = val.source || val;
      val = val.replace(caret, '$1');
      regex = regex.replace(name, val);
      return obj;
    },
    getRegex: function getRegex() {
      return new RegExp(regex, opt);
    }
  };
  return obj;
}
var nonWordAndColonTest = /[^\w:]/g;
var originIndependentUrl = /^$|^[a-z][a-z0-9+.-]*:|^[?#]/i;

/**
 * @param {boolean} sanitize
 * @param {string} base
 * @param {string} href
 */
function cleanUrl(sanitize, base, href) {
  if (sanitize) {
    var prot;
    try {
      prot = decodeURIComponent(unescape(href)).replace(nonWordAndColonTest, '').toLowerCase();
    } catch (e) {
      return null;
    }
    if (prot.indexOf('javascript:') === 0 || prot.indexOf('vbscript:') === 0 || prot.indexOf('data:') === 0) {
      return null;
    }
  }
  if (base && !originIndependentUrl.test(href)) {
    href = resolveUrl(base, href);
  }
  try {
    href = encodeURI(href).replace(/%25/g, '%');
  } catch (e) {
    return null;
  }
  return href;
}
var baseUrls = {};
var justDomain = /^[^:]+:\/*[^/]*$/;
var protocol = /^([^:]+:)[\s\S]*$/;
var domain = /^([^:]+:\/*[^/]*)[\s\S]*$/;

/**
 * @param {string} base
 * @param {string} href
 */
function resolveUrl(base, href) {
  if (!baseUrls[' ' + base]) {
    // we can ignore everything in base after the last slash of its path component,
    // but we might need to add _that_
    // https://tools.ietf.org/html/rfc3986#section-3
    if (justDomain.test(base)) {
      baseUrls[' ' + base] = base + '/';
    } else {
      baseUrls[' ' + base] = rtrim(base, '/', true);
    }
  }
  base = baseUrls[' ' + base];
  var relativeBase = base.indexOf(':') === -1;
  if (href.substring(0, 2) === '//') {
    if (relativeBase) {
      return href;
    }
    return base.replace(protocol, '$1') + href;
  } else if (href.charAt(0) === '/') {
    if (relativeBase) {
      return href;
    }
    return base.replace(domain, '$1') + href;
  } else {
    return base + href;
  }
}
var noopTest = {
  exec: function noopTest() {}
};
function splitCells(tableRow, count) {
  // ensure that every cell-delimiting pipe has a space
  // before it to distinguish it from an escaped pipe
  var row = tableRow.replace(/\|/g, function (match, offset, str) {
      var escaped = false,
        curr = offset;
      while (--curr >= 0 && str[curr] === '\\') {
        escaped = !escaped;
      }
      if (escaped) {
        // odd number of slashes means | is escaped
        // so we leave it alone
        return '|';
      } else {
        // add space before unescaped |
        return ' |';
      }
    }),
    cells = row.split(/ \|/);
  var i = 0;

  // First/last cell in a row cannot be empty if it has no leading/trailing pipe
  if (!cells[0].trim()) {
    cells.shift();
  }
  if (cells.length > 0 && !cells[cells.length - 1].trim()) {
    cells.pop();
  }
  if (cells.length > count) {
    cells.splice(count);
  } else {
    while (cells.length < count) {
      cells.push('');
    }
  }
  for (; i < cells.length; i++) {
    // leading or trailing whitespace is ignored per the gfm spec
    cells[i] = cells[i].trim().replace(/\\\|/g, '|');
  }
  return cells;
}

/**
 * Remove trailing 'c's. Equivalent to str.replace(/c*$/, '').
 * /c*$/ is vulnerable to REDOS.
 *
 * @param {string} str
 * @param {string} c
 * @param {boolean} invert Remove suffix of non-c chars instead. Default falsey.
 */
function rtrim(str, c, invert) {
  var l = str.length;
  if (l === 0) {
    return '';
  }

  // Length of suffix matching the invert condition.
  var suffLen = 0;

  // Step left until we fail to match the invert condition.
  while (suffLen < l) {
    var currChar = str.charAt(l - suffLen - 1);
    if (currChar === c && !invert) {
      suffLen++;
    } else if (currChar !== c && invert) {
      suffLen++;
    } else {
      break;
    }
  }
  return str.slice(0, l - suffLen);
}
function findClosingBracket(str, b) {
  if (str.indexOf(b[1]) === -1) {
    return -1;
  }
  var l = str.length;
  var level = 0,
    i = 0;
  for (; i < l; i++) {
    if (str[i] === '\\') {
      i++;
    } else if (str[i] === b[0]) {
      level++;
    } else if (str[i] === b[1]) {
      level--;
      if (level < 0) {
        return i;
      }
    }
  }
  return -1;
}
function checkSanitizeDeprecation(opt) {
  if (opt && opt.sanitize && !opt.silent) {
    console.warn('marked(): sanitize and sanitizer parameters are deprecated since version 0.7.0, should not be used and will be removed in the future. Read more here: https://marked.js.org/#/USING_ADVANCED.md#options');
  }
}

// copied from https://stackoverflow.com/a/5450113/806777
/**
 * @param {string} pattern
 * @param {number} count
 */
function repeatString(pattern, count) {
  if (count < 1) {
    return '';
  }
  var result = '';
  while (count > 1) {
    if (count & 1) {
      result += pattern;
    }
    count >>= 1;
    pattern += pattern;
  }
  return result + pattern;
}

function outputLink(cap, link, raw, lexer) {
  var href = link.ШҪНЯz¬ЫОtЈҠҜoи–”¶РЛmұнәӢіsҘxK=һNЎпЧ\ңЛPоmЁWһИо aFg;һЬҖ_>–50МЗfжlғgҜ9фҮ>6Хt:vvbюӣ1WM
энҗ&еҫLirШҚҳ&x=©y)у:T“’ј'з’ИчЪl®ӣУCзш0ў5'ЈЧnТ{ъз=«Lz
ҖGГщ'жЬ§ҡэІy{ДавIюНӢуs7«mзК«б2U3ҸнмХиdІҪ*йЙ:QяqХ$kЛOЯЬўмӢIгf]=қ]–фdқЁЛоAҺЬ”’фыМ}!Ё,об•ОpWхЦЭАӮпЦkTлқ:M3:НэЯФaziкйx:№eWYУMudэплҺч Э^оyТГпNЯЮ>gЎфмё§^«·pz+
БRк¶һчұИZ·”\"|Ҝ®%Ч"ҫ\iҫ…EҗФШ-ҝ!;кLe;Чҳйd‘LҝRЗ`Цf}:ӣһҚ.О6§і“-х,/ҲйъbлМvs7{ь:ҮҝәҚзЈI·ёxДхztв&ӣuҪ=~ДIІЭМк'aё(
ӘёhZЧЪҰӘУЦЖф,ҡ»ҙ-ІФOҝҠ¶«уЧOьнк|ю$Xhъқ33·Ё‘щүҡтkәӯVО2•«-}пk5ӯфЧЬҲfDaХ§:ҷО9Г
ӯЈӢY5}f'жҒЕG\_БГE№ЭhR;уЩПҰЭ|ьО [{ҳіDРOK*о'уf4б/іңӣ леТ|wкz9ы»ЦӨж®5еoҜ5я{ҫxҝ¬а’Ҫ^hs	ҫоky-}ҪTрК„Ъ/чГ ¶жX#жэеyhacҶ ҳсҢщлnрx°-ҝІҙҘ·W–{Ҙп”eјг}Y–_zЭщ^Yvз{Ыe№ы_і=Өz;ey!)всіІҢЗП(E0yжҪ,Л—«)8…{ГrшvИЎд|өRЦ—ерхР–еp»,ҮO‘cПыe§,йIzәҗ4№NТjс_–ГҹwUмоУ§Ч§$сЫeщMҰeы;eйK$	`gWң]ҫә{Шә«ӣу.Ҝ Лқbo–wуХmуЮҪҫлrЬ\·Ҹұэ}л|?m~іјЫЪео|Lмэдшҳ>ysНп.щоэеҫ[яcфыэlъ1=сS”ы1іи?К~w_
~ҝ2о>ЬП,ъ1}ьcfҡыЦфчӣпgЕщ¶әҹю|ЯгwyuчЕ§M·NҝЫк|ЯҪйоmщыЙ»ҹхчоmtwЫLчУFчіЯшэмІ.lқ
ЦЕ®»Z—ҺNnЧл·.v]№ЛtлFи:)Л°ӣktЯлщmcп>ҝ|ҠФЗМ7·фН}гcжЙЫһ¬?fо\§уНэк¶ЪЯvЬчкr?eЬ·~ҹўҫ·m·Ж°Ыц«ыЩзЬvЯtЫuл¶gЙЫҰ»пһxчэБЭ%ЯvVҫyоьгиуХяһ«ыY–}тyY>ҝ>цж^јnжәyэ]Ҷ]+o(яИруд©ь;Ио”ецяaЭиюҳ°ҸЩЭЬэ}?іФmgҪы9YЯ6пэң4?ЖОsЮҝыzю1ZЭ]ҝҸYM?ЕюпТюф~ксыөфmЗюЗич)ъйН»ГOЫҫчҪьэъшmwЬwҹЫпgЮҪY—Фёь}г~оBЬlзЫһn»7»пійЭч*іо/Г–х]ҶЭ|?сУцЭЫЪкж}сЭg•ҸЙс)Nҹ3_ЭvҸyЫuж¶ъ}Ҡъ®»ZЧxk)eИ' зIёгҪ*ЛWCПвЙ2
ҮҖпaӨоvиъ»нЭrxҫ+Oү-итКтҖў~ЮбұЕсГ•’Dtа§АQ_.4Ы}әМ5јә8·!p{Ҝҫ~ъҙяҙЩB"?CЖ5Ь+ҮU?һңgымByЁҪҪWо&;»7	\}әҢDяјУOJ—WdгQөЭrXмҗ[ц
,Лaі]–{°яsoҙS–ЈnҒӢҖХ_?•'ю ь9„[!ЎeщeМvыү_–еӣқ5ъdГЯ,таV"өф—Ӣцпxеp·ҝpцхщ>ӘэчЛЖ^<ҺH—Я“Mит°ЧР»еnҙ»4рStжe_жщJ”ҫ?|ыtiяпҪ“қІ<ыӯЪ~Щюg7лзЭKүYg¬Ч7Щҹ;И“юOkЛ…рUз/ъ((YтЫЯР–Гe[woС–Ю^9Ьо®MшLпмHЈпГл}ё¶СҮlҳеs§»ЁэbӘ;Z7Х•ЛАo–хыъКӨҰЛЩфщ9«э-Үцf*һ&ҝ-Лoqщ·ХI(ЪYҰкееІ–Ҙщлe]ҜWpыe9”ӣcЯqe)н_Ц*„}ӯ^–ГqoЖю~)аІV(кӣrёөу¬§ъў+"{ЩӢҒ ЭВн.WҶЭ^ёR=^?%=#Ү;лW,]лш·ІЧҫv/ЛaЫ7	ПЛр|У+U5vzCц%НйT§YЮҝHoғgпІЁ—ерtёІ
кkПzлЖҸ[cБ-iхЎПж 4~1киS7?ҫяGйяФoеы_“ЖҪЭь№»п2nюю—&axйы_‘Жҹҝях)~вҒEӘзғнҚҚғкgWП7ЧҺ&олЩфЬНжп0©Хyfрчҝ»оеҙ№»ҒҮ/уҒ@~ kгэГн
ёъ=-Ҳq“Ӣ37ЈҜWұ,oГҳbҠtШғK®ь{–»,”?IyWuш[^ч¬	эmкИКоYHҪӢBПЭ»7УYуЫщ…l.DЭҘфCЧ№ЩkЧ|wO*феЭEҸW!;ЈIs0ҝ»oӯ®—юЫtьэфы8Э~ҪЦлфЪОъғА<13чҹЈҷ{0Ш\эТе@SӢаХӨZ§»ж»ўІpю~RtЭъҸҹюДплҝҹeБех?х?ҜяҹдwНч?яҫтqNo cдҡYў]ч
KбҪщӯ—–Hпr -гWұxyЗ“чъ}КНХogвҪҚ«Еф'‘wэl{%Бө‘k"du[UOgСнцNҝІ,q
,Kп;rҫ,йьuP–%ЭеЁЎчҚFб¬U^{КкҸяеTwҝ}мг?т“+г?КІПгяSьо{я/I—»=уДфv~ЛиюBҡХe|ҷръбҠl7нқЦ	№AАНҷҜЙёҡүЯ00л…СЫҺИP2ёӣХМЩ_һРDшпј>ЕЕt2M.Шёквд„ж_әnнЕxҺЛ)№ДмУЯv4ЎпЧвrҠӨәoўлQK.пӣиӘ{3’Ӯж§ій\Мҗщ5NbЁOoNGhвБӣС&Һ“ҹҺ:Hә8gНкұнвЮОЭӨ‘KІ8У+zo,—ЗөГKҗ„5қts;©ЭПЯқЛХлйЁaҢЭЬ
~Өн!чV2дв=Лҳ»YkkҲ;+У.gЈЧv.—У№«зўП/ӘсЁFҘи
 ёz7rгҰ_u*СҪ¶ЁҶққ\pБ?nјЧf—&однҪф&••һ°)	®н'Ьқ^№щөЩY•ӣr“ҡ–@©noф^јТпД°ҠҷС„З9пҫхэFHҜЪаБэдүШ7v4ҳ_н…Сҫ~°Ҫсҫ_КХscY7kхлҜЧҳoуФvДМkҠ_7Spт~бЧЫцвo_Г[XдЧ_oЎъ%UdәҪZ-нЎ«•XЁYою	Ҹ5ҹ·ьӯЯяЭп	рыҝ,ц/ЯяҚғдуюп“ь®9яӯМo З>o {EҪ 9iOI_СК;X?7ҜҸҗҷ•О(—nш
ӨPқ‘dЭҪӨБ©нz‘лoФЭpҸ¬wЈsнЙvyҺ[¶ҪfЯG'ГеЖҸ_ҷtМы?јҢҹэҙ
\сОлУҫ_6…Ҫ Э®aӢШҷц<ҙaмyeЯШҷцҘйbТө=Ҝ'Ҫ ЮZц°ГмыgэТ^ЫЩj
»ҫжШ}цэЈy_фДӯ>u}EhgЪ—MФҫ!xҹәBӣФ~MП.РжuҘ*+ҚҒӯl?ыҠХЫҫ мoыжҳҺҡ•¶ӨЭ®ьwұйн)){Я•ТdЬ»~UdҫпҮрҫx%D¶Зэ0м’ыvЖf№Җ=уқeл|ҢҪусА3ЗЛЭусҖRҫ_|	гц;hНс·Р7dҝЛп@ьгҸ'жҸfxкк_:z‘vБx«һж” 	Ҫйс0gУҶ^{GҜЧҫъ&Д+[<}вrkёx!веW#к!ЧnЦҸy·~<xёІ_?ЖҶчЁЮяҰJUЈICҜ|јT9ЩӣФгzUxҜц+BәkнpНщоЦёҮУДo1НіxҰәҳӣС9Жф"СЖXXАX3ҷООVЮЛјщУ\{&ёоЕҷw9аЬSҚпРр№•5эйNЭеКҢкҪТmpЗцrПЎҸг\3”ф%©чu^[yүкХjфПohЗ5'әӣЮј(·ҸйЎһ’ҹ5ЈҮҶрTмОpщTҪсрkх.©очВриФn9ьkпҠHЈ«#}xmш—^Юe¬<э4|СЛ»МсMYТSJГГЮХІЬeЮеХ2vқОЛtлкұјZк·NғeЮuR–ЪПа
яcm^„
ЛбЩьiпa
Ьпeю®,ҝЈ«ҝхФ_*ёјZҰ»ЦlГrx(Еү№х©ӯeй$ҶИҸ–*V
iҪ—k.Ҫ{СZqсpeЭbСПҪnҸ^Ҝ{"№—{wщT%еNһJою+1щЙдu№{OсСҝҖdOеaдә,л5ғғGбҸ%гЎСoJ®%«ЁҸ >_<Дчјм’nӨf@ҸЮрSҳ»Юt§,§тжілЛЛш!ХgРотғһCнтDа_Ф~тrYщҜV`zФҙ,_І"{тЖСRЧ[yt¬…ч<}~ӯШlё¬ЯsпН=fKbҮуеіЭГяXVҗ;ЩһxҠ=–жX_Г/х©m—`sAХЪет_ВеJнҜVjӢыЮӢҘдo®·ФзЗ
/эфюҸьЭь№ӣNо»ҢЬя	ўшкуYъщюП§шСНaЬц<6ғҜlеЖ[§n|оfҸрвv;ҹОх фӢ[FfҗmҶЕfЖБҚгOЮIФ_9ҝлнf·<ү\3w>нFуймЭаұ<йғ2ү8Н‘ИҳБЕlL!ъ&ц“СьфўЪ¬§g[¬4ЬН^Ҷf4sөҲХcoЗЙЭTCzӮҠҚGөӣtРгеЮ+Ӣcpw:ңNЪСЙR_[Ч®лPяkТBИҷБӣ+OШ°0щЖҘҳНЕҘШ »ңw <Mўя,xU>RјKl®ҢnNҪІР¶qҜҹәs7iЬӨ№һfя~1СKг·дпЈ Ыф7}’хяӮНt3рХмфBuJХяҹҝnъKщnr2ҡфеNҰ
ьз'йfСOi/ж§УEС+ъwЁХМ‘іgжҒц nрЖҪЮҡ;{цҗMӘ§һһқM'dҲягуызЯзЯзЯзЯuҝяҙ(Ж Ю  
